examples/common/datt-config/DattConfig.cpp

```diff
@@ -0,0 +1,29 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <DattConfig.h>
+
+__attribute__((weak)) const chip::Credentials::AttestationTrustStore * GetAttestationTrustStore()
+{
+    return chip::Credentials::GetTestAttestationTrustStore();
+}
+
+__attribute__((weak)) chip::Credentials::DeviceAttestationCredentialsProvider * GetAttestationCredentialsProvider()
+{
+    return chip::Credentials::Examples::GetExampleDACProvider();
+}
```

Comment:
This PR doesn't compromise the example. This PR only adds a new possibility for developers to re-use the current example with their own DAC Provider and TrustedRoot List, by simply adding 2 method definitions that will override those new weak symbols.
In this case, lighting-app will continue to build and act as usual.

Reply:
I think Tennessee's point, though, is that the proper way to use a platform DAC provider is not to override the example one, but to just set the appropriate credentials provider using the call to SetDeviceAttestationCredentialsProvider, which already exists. The example credentials provider is being used because it is already being explicitly set.